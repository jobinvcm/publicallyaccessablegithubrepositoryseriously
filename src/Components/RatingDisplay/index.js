import React from 'react';
import styled from 'styled-components';
import propTypes from 'prop-types';

const StyledSpan = styled("span")`
  background-color: ${props => props.ratingColor};
  display: block;
  padding: 3px 9px;
  font-size: 12px;
  line-height: 14px;
  color: #fff;
  font-weight: bold;
  border-radius: 8px;
`;

const RatingDisplay = props => {
    const { lowRatingColor, midRatingColor, highRatingColor, rating } = props;
    let stdRating = rating * 10;
    let ratingColor = midRatingColor;
    if (stdRating < 30) {
        ratingColor = lowRatingColor;
    } else if (stdRating > 80) {
        ratingColor = highRatingColor;
    }
    
    return <StyledSpan ratingColor={ratingColor}>{stdRating}%</StyledSpan>
};

RatingDisplay.propTypes = {
  lowRatingColor: propTypes.string,
  midRatingColor: propTypes.string,
  highRatingColor: propTypes.string,
  rating: propTypes.number.isRequired,
};

RatingDisplay.defaultProps = {
  lowRatingColor: '#D1225B',
  midRatingColor: '#4902A3',
  highRatingColor: '#01D277',
}

export default RatingDisplay;