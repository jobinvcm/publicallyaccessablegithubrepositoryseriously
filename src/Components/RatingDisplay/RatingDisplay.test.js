import React from 'react';
import { shallow } from 'enzyme';
import RatingDisplay from '.';

describe('<RatingDisplay />', () => {
  it('renders RatingDisplay component', () => {
    const wrapper = shallow(<RatingDisplay rating={3.5} />);
    expect(wrapper).toMatchSnapshot();
  });
});
