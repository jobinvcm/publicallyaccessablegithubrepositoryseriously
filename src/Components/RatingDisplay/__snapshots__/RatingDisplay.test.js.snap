// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatingDisplay /> renders RatingDisplay component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RatingDisplay
    highRatingColor="#01D277"
    lowRatingColor="#D1225B"
    midRatingColor="#4902A3"
    rating={3.5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        35,
        "%",
      ],
      "ratingColor": "#4902A3",
    },
    "ref": null,
    "rendered": Array [
      35,
      "%",
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  background-color: ",
          [Function],
          ";
  display: block;
  padding: 3px 9px;
  font-size: 12px;
  line-height: 14px;
  color: #fff;
  font-weight: bold;
  border-radius: 8px;
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          35,
          "%",
        ],
        "ratingColor": "#4902A3",
      },
      "ref": null,
      "rendered": Array [
        35,
        "%",
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  background-color: ",
            [Function],
            ";
  display: block;
  padding: 3px 9px;
  font-size: 12px;
  line-height: 14px;
  color: #fff;
  font-weight: bold;
  border-radius: 8px;
",
          ],
        },
        "displayName": "styled.span",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "span",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
